### PR TITLE
Address additional restore changes feedback

### PR DESCRIPTION
**Pull request by dibarbet**
Addressing post merge feedback on https://github.com/dotnet/roslyn/pull/70588

### CODE REVIEW

File: src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/LanguageServer/Handler/Restore/RestorableProjectsHandler.cs [hunk 1]
```diff
@@ -31,23 +31,20 @@ public Task<string[]> HandleRequestAsync(RequestContext context, CancellationTok
...
+        //   2.  Removes projects with duplicate file paths (for example multi-targeted projects).  They all get restored
+        //       together by file path.
+        var projects = new SortedSet<string>();
         foreach (var project in context.Solution.Projects)
         {
             // To restore via the dotnet CLI, we must have a file path and it must be a .NET core project.
             if (project.FilePath != null && projectTargetFrameworkManager.IsDotnetCoreProject(project.Id))
             {
-                projectsBuilder.Add(project.FilePath);
+                _ = projects.Add(project.FilePath);
```
Comment (jasonmalinowski) [inline src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/LanguageServer/Handler/Restore/RestorableProjectsHandler.cs:44]: Why the explicit throwaway?

File: src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/LanguageServer/Handler/Restore/RestoreHandler.cs [hunk 2]
```diff
@@ -49,13 +49,18 @@ private async Task RestoreAsync(ImmutableArray<string> pathsToRestore, BufferedP
     {
         foreach (var path in pathsToRestore)
         {
-            var arguments = $"restore \"{path}\"";
+            var arguments = new string[] { "restore", path };
```
Comment (jasonmalinowski) [inline src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/LanguageServer/Handler/Restore/RestoreHandler.cs:52]: Didn't want to inline with collection expressions here?

File: src/Features/LanguageServer/ProtocolUnitTests/Hover/HoverTests.cs [hunk 3]
```diff
@@ -419,6 +419,37 @@ public async Task TestGetHoverAsync_UsingMarkupContentDoesNotEscapeCode(bool mut
...
+";
+
+            var results = await RunGetHoverAsync(
+                testLspServer,
+                expectedLocation).ConfigureAwait(false);
+            Assert.Equal(expectedMarkdown, results.Contents.Value.Fourth.Value);
+        }
+
+        [Theory, CombinatorialData, WorkItem("https://github.com/dotnet/vscode-csharp/issues/6577")]
+        public async Task TestGetHoverAsync_UsesCodeFencesInAwaitReturn(bool mutatingLspWorkspace)
```
Comment (jasonmalinowski) [inline src/Features/LanguageServer/ProtocolUnitTests/Hover/HoverTests.cs:432]: Mixed up the PRs?

**Review by jasonmalinowski — APPROVED** [review]

File: src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/LanguageServer/Handler/Restore/RestoreHandler.cs (hunk 2 above)
Comment (dibarbet) [reply src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/LanguageServer/Handler/Restore/RestoreHandler.cs:52]: There appear to be a bunch of formatting issues with collection expressions.  Every time I try to use them I get formatting warnings (even with 'correct' formatting).  I'll file a bug on that one sec.

**Pull request merged**